build-ps/test-build.sh

```diff
@@ -1,61 +1,81 @@
 #!/bin/bash
 
 function check_libs {
-    for elf in $(find $1 -maxdepth 1 -exec file {} \; | grep 'ELF ' | cut -d':' -f1); do
-        echo $elf
-        ldd $elf | grep "not found"
+    local elf_path=$1
+
+    for elf in $(find $elf_path -maxdepth 1 -exec file {} \; | grep 'ELF ' | cut -d':' -f1); do
+        echo "$elf"
+        ldd "$elf"
     done
     return
 }
 
 function prepare {
     CURDIR=$(pwd)
+    TMP_DIR="$CURDIR/temp"
+
+    mkdir -p "$CURDIR"/temp
+    mkdir -p "$TMP_DIR"/dbdeployer/tarball "$TMP_DIR"/dbdeployer/deployment
+
+    TARBALLS=""
+    for tarball in $(find . -name "*.tar.gz"); do
+        TARBALLS+=" $(basename $tarball)"
+    done
 
-    mkdir -p $CURDIR/temp
-    TMP_DIR=$CURDIR/temp
-    
-    TARBALL=$(basename $(find . -name "*glibc2.12.tar.gz" | sort))
+    DIRLIST="bin lib lib/private lib/plugin lib/mysqlrouter/plugin lib/mysqlrouter/private"
 }
 
 function install_deps {
     if [ -f /etc/redhat-release ]; then 
-        yum install -y perl-Time-HiRes perl numactl numactl-libs libaio libidn || true
+        yum install -y wget perl-Time-HiRes perl numactl numactl-libs libaio libidn || true
     else
-        apt install -y perl numactl libaio-dev libidn11 || true
+        apt install -y wget perl numactl libaio-dev libidn11 || true
     fi
+    wget -c https://github.com/datacharmer/dbdeployer/releases/download/v1.52.0/dbdeployer-1.52.0.linux.tar.gz -O - | tar -xz
+    mv dbdeployer*.linux /usr/local/bin/dbdeployer
+    export PATH=$PATH:/usr/local/bin
 }
 
 main () {
-
-    DIRLIST="bin lib lib/private lib/plugin lib/mysqlrouter/plugin lib/mysqlrouter/private"
-
     prepare
+    for tarfile in $TARBALLS; do
+        echo "Unpacking tarball: $tarfile"
+        cd "$TMP_DIR"
+        tar xf "$CURDIR/$tarfile"
+        cd "${tarfile%.tar.gz}"
 
-    echo "Unpacking tarball"
-    cd $TMP_DIR
-    tar xf $CURDIR/$TARBALL
-    cd ${TARBALL%.tar.gz}
-
-    echo "Checking ELFs for not found"
-    for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
-    done
+        echo "Building ELFs ldd output list"
+        for DIR in $DIRLIST; do
+            if ! check_libs "$DIR" >> "$TMP_DIR"/libs_err.log; then
```

Comment:
please add []

Reply:
Expected comparison operator (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.shellcheck(SC1072)